ReactAndroid/src/main/java/com/facebook/react/modules/location/LocationModule.java

```diff
@@ -112,6 +116,50 @@ private static LocationOptions fromReactMap(ReadableMap map) {
    */
   @ReactMethod
   public void getCurrentPosition(
+      final ReadableMap options,
+      final Callback success,
+      final Callback error) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      final PermissionsModule perms = getReactApplicationContext().getNativeModule(PermissionsModule.class);
+      perms.checkPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(new Callback() {
+        @Override
+        public void invoke(Object... args) {
+          boolean hasPermission = (boolean) args[0];
+
+          if (!hasPermission) {
+            perms.requestPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(new Callback() {
+              @Override
+              public void invoke(Object... args) {
+                String result = (String) args[0];
+                if (result == "granted") {
+                  loadCurrentPosition(options, success, error);
+                } else {
+                  error.invoke(PositionError.buildError(PositionError.PERMISSION_DENIED, "Location permission was not granted."));
+                }
+              }
+            }, new Callback() {
```

Comment:
can you extract these inline Callback ctors to named variables so the code is a bit easier to follow?

Reply:
Done. I will squash the two commits once it has been approved.